contracts/potlock/src/potlock_admin_interactions.rs

```diff
@@ -0,0 +1,100 @@
+};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+pub type ProjectPercentage = MultiValue2<usize, u64>;
+const MAX_PERCENTAGE: u64 = 10_000; // 100%
+
+#[multiversx_sc::module]
+pub trait PotlockAdminInteractions:
+    potlock_storage::PotlockStorage
+    + multiversx_sc_modules::only_admin::OnlyAdminModule
+    + potlock_setup::PotlockSetup
+{
+    #[only_admin]
+    #[endpoint(acceptPot)]
+    fn accept_pot(&self, potlock_id: PotlockId) {
+        self.require_potlock_exists(potlock_id);
+        let fee_amount = self.fee_amount().get();
+
+        self.fee_amount_accepted_pots()
+            .update(|amount| *amount += fee_amount);
+        let mut accepted_potlock = self.potlocks().get(potlock_id);
+        accepted_potlock.status = Status::Active;
+        self.potlocks().set(potlock_id, &accepted_potlock);
+        self.fee_pot_proposer(potlock_id).clear();
+    }
+
+    #[only_admin]
+    #[endpoint(removePot)]
+    fn remove_pot(&self, potlock_id: PotlockId) {
+        self.require_potlock_exists(potlock_id);
+
+        let pot_proposer = self.fee_pot_proposer(potlock_id).take();
+        let fee_pot_payment = EsdtTokenPayment::new(
+            self.fee_token_identifier().get(),
+            0u64,
+            self.fee_amount().get(),
+        );
+
+        self.send()
+            .direct_non_zero_esdt_payment(&pot_proposer, &fee_pot_payment);
+        self.potlocks().clear_entry(potlock_id);
+    }
+
+    #[only_admin]
+    #[endpoint(acceptApplication)]
+    fn accept_application(&self, project_id: ProjectId) {
+        self.require_project_exists(project_id);
+        let mut accepted_projects = self.projects().get(project_id);
+        accepted_projects.status = Status::Active;
+        self.projects().set(project_id, &accepted_projects);
+    }
+
+    #[only_admin]
+    #[endpoint(rejectDonation)]
+    fn reject_donation(&self, potlock_id: PotlockId, user: ManagedAddress) {
+        self.require_potlock_exists(potlock_id);
+        let opt_fee_pot_payments = self.pot_donations(potlock_id).get(&user);
+
+        require!(opt_fee_pot_payments.is_some(), "No donation for this user");
+        let fee_pot_payments = unsafe { opt_fee_pot_payments.unwrap_unchecked() };
+
+        self.send()
+            .direct_non_zero_esdt_payment(&user, &fee_pot_payments);
+        self.pot_donations(potlock_id).remove(&user);
+    }
+
+    #[only_admin]
+    #[endpoint(distributePotToProjects)]
+    fn distribute_pot_to_projects(
+        &self,
+        potlock_id: PotlockId,
+        project_percentage: MultiValueEncoded<ProjectPercentage>,
+    ) {
+        self.require_potlock_exists(potlock_id);
+        let pot_donations = self.pot_donations(potlock_id);
+
+        for pp in project_percentage {
+            let (project_id, percentage) = pp.into_tuple();
+            let mut output_payments = ManagedVec::new();
+            for (_, donation) in pot_donations.iter() {
```

Comment:
There may be a few problems with this way of implementing (as Dorin also suggested), mostly concerning gas usage. If a malicious user donates values from 1000 addresses, the potlock will be blocked, with an outOfGas error.
Maybe rethink the architecture of the storages. One option would be instead of a mapMapper, to save under the potlock_id and user keys, a singleValueMapper of PaymentsVec. This allows you to quickly go to a specific user for donations and also, this endpoint would control the gasCost by clearly providing a list of users, besides the potlockId.